src/v/cloud_storage/partition_manifest.cc

```diff
@@ -2574,12 +2574,21 @@ void partition_manifest::process_anomalies(
                == _spillover_manifests.end();
     });
 
+    // TODO: check if the offset range is covered by some other segment
     auto first_kafka_offset = full_log_start_kafka_offset();
     auto& missing_segs = _detected_anomalies.missing_segments;
     erase_if(missing_segs, [&first_kafka_offset](const auto& meta) {
         return meta.next_kafka_offset() <= first_kafka_offset;
     });
 
+    // TODO: check if the segment still exists
+    auto& segment_meta_anomalies
+      = _detected_anomalies.segment_metadata_anomalies;
+    erase_if(
+      segment_meta_anomalies, [&first_kafka_offset](const auto& anomaly_meta) {
+          return anomaly_meta.at.next_kafka_offset() <= first_kafka_offset;
```

Comment:
It is also possible that the segment is scanned and replaced concurrently. In order to filter out such segments you need to take into account other parameters (size, base/committed offsets). The segment might be in the `_replaced` list of not (if the cleanup happened concurrently).

Reply:
You're right. The commit with the title from below does what you're suggesting and adds a test for such situations:
> cloud_storage: improve manifest anomaly filtering